path-walk.c

```diff
@@ -84,6 +84,10 @@ static int add_children(struct path_walk_context *ctx,
 		if (S_ISGITLINK(entry.mode))
 			continue;
 
+		/* If the caller doesn't want blobs, then don't bother. */
+		if (!ctx->info->blobs && type == OBJ_BLOB)
```

Comment:
But we need to walk trees to find the blobs. We won't call the callback for trees if `ctx->info->trees` is false.

Reply:
Oh, I missed that it is [a _positive_ guard](https://github.com/git-for-windows/git/pull/5171/files#diff-e8488c9726c2e32c86422224262bb3d0f95d9a3ec56cb9d7f201d3b121d61d80R186-R189): we're only adding trees there if `ctx->info->trees` is non-zero (and likewise blobs if `ctx->info->blobs` is non-zero). So that already takes care of this, and I mistook the hunk on which I commented for _being_ that guard. Sorry for the noise!